packages/teams-js/src/internal/utils.ts

```diff
@@ -36,13 +37,16 @@ function validateHostAgainstPattern(pattern: string, host: string): boolean {
   return false;
 }
 
+const validateOriginLogger = getLogger('validateOrigin');
+
 /**
  * @internal
  * Limited to Microsoft-internal use
  */
 export function validateOrigin(messageOrigin: URL): boolean {
   // Check whether the url is in the pre-known allowlist or supplied by user
   if (!isValidHttpsURL(messageOrigin)) {
+    validateOriginLogger('Origin %s is invalid because it is not using https protocol', messageOrigin);
```

Comment:
>not using https protocol [](http://example.com/codeflow?start=61&length=24)

Perhaps this is pointless because "not https" basically means "http" but you could consider logging `messageOrigin.protocol`

Reply:
I don't think it serves much of a purpose, but it also doesn't hurt and is easy to add so I did